src/gt4py/next/otf/workflow.py

```diff
@@ -128,3 +128,41 @@ def __call__(self, inp: StartT) -> EndT:
 
     def chain(self, step: Workflow[EndT, NewEndT]) -> CombinedStep[StartT, EndT, NewEndT]:
         return CombinedStep(first=self, second=step)
+
+
+@dataclasses.dataclass(frozen=True)
+class CachedWorkflow(Generic[StartT, EndT]):
```

Comment:
Why is this not a Step subclass? It lives on the same level, has the same interoperability etc...

Either case, the current naming is `Workflow` for the concept and `XyzStep` for all the concrete implementations (Any workflow can be a step in a super-workflow anyway). So naming this Concrete implementation CachedWorkflow makes it potentially confusing.